src/block_cache/block_chunkcache.c

```diff
@@ -845,6 +846,43 @@ __wt_chunkcache_remove(
     return (0);
 }
 
+/*
+ * __wt_chunkcache_unpin_old_versions --
+ *     Unpin the old versions of newly added chunks so eviction can remove them.
+ */
+void
+__wt_chunkcache_unpin_old_versions(WT_SESSION_IMPL *session, const char *sp_obj_name)
+{
+    WT_CHUNKCACHE *chunkcache;
+    WT_CHUNKCACHE_CHUNK *chunk, *chunk_tmp;
+    int i;
+
+    chunkcache = &S2C(session)->chunkcache;
+
+    /* Optimization: check if the file contains objects in the pinned list, otherwise skip. */
+    if (__chunkcache_should_pin_chunk(session, sp_obj_name)) {
+        /* Loop through the entire chunkcache and search for matching objects from the file and
+         * clear the pinned flag. */
+        for (i = 0; i < (int)chunkcache->hashtable_size; i++) {
```

Comment:
`i` should probably be a `size_t` (or whatever) so we can drop the cast.

Reply:
addressed here [de67e9a](https://github.com/wiredtiger/wiredtiger/pull/9730/commits/de67e9a805f3926a1d3384e7dd1c81db5411c3db)